### PR TITLE
dbeaver/dbeaver#20528 Initial commit for my case insensitive feature.

### LINKED ISSUE #20528: Order case insensitive as an option at the Client Side
**Is your feature request related to a problem? Please describe.**

**Describe the solution you'd like**

**Describe alternatives you've considered**

**Additional context**

### PR DESCRIPTION
**Pull request by vijayap20**
Order case insensitive as an option at the Client Side #20528[](url)

### CODE REVIEW

**Comment by E1izabeth** [comment]
Thank you for contribution.
We don't want to add such option. Let's just change the behaviour of the existing `Always on client` setting. The sorting should always be case-insensitive.

Also only strings should be compared in such a way - it doesn't make sense to do it for numbers.
And pay attention on different locales. It should correctly work with non-latin languages.

**Comment by E1izabeth** [comment]
Please delete all changes that are no longer in use

File: plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/ResultSetModel.java [hunk 1]
```diff
@@ -1123,7 +1123,7 @@ public void resetOrdering() {
                     }
                     Object cell1 = getCellValue(new ResultSetCellLocation(binding, row1));
                     Object cell2 = getCellValue(new ResultSetCellLocation(binding, row2));
-                    result = DBUtils.compareDataValues(cell1, cell2);
+                    result = DBUtils.compareDataValues((Object) cell1.toString().toLowerCase(), (Object) cell2.toString().toLowerCase());
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/ResultSetModel.java]: Please do such comparison only for string types. It makes no sense to do it for numbers or binary objects.

**Review by E1izabeth — APPROVED** [review]

File: plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/ResultSetModel.java [hunk 2]
```diff
@@ -1123,7 +1123,12 @@ public void resetOrdering() {
                     }
                     Object cell1 = getCellValue(new ResultSetCellLocation(binding, row1));
                     Object cell2 = getCellValue(new ResultSetCellLocation(binding, row2));
-                    result = DBUtils.compareDataValues(cell1, cell2);
+                    if(cell1 instanceof String && cell2 instanceof String) {
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/ResultSetModel.java]:
```suggestion
                    if (cell1 instanceof String && cell2 instanceof String) {
```

File: plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/ResultSetModel.java [hunk 3]
```diff
@@ -1123,7 +1123,12 @@ public void resetOrdering() {
...
+                    	result = DBUtils.compareDataValues((Object) cell1.toString().toLowerCase(), (Object) cell2.toString().toLowerCase());                    	
+                    }else {
```
Comment (ShadelessFox) [inline plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/ResultSetModel.java]:
```suggestion
                    } else {
```

Comment (vijayap20) [reply plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/ResultSetModel.java]: resolved

**Review by ShadelessFox — APPROVED** [review]

**Comment by E1izabeth** [comment]
Passed to QA team.

**Comment by Matvey16** [comment]
Verified

File: plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/ResultSetModel.java [hunk 4]
```diff
@@ -1123,7 +1123,12 @@ public void resetOrdering() {
                     }
                     Object cell1 = getCellValue(new ResultSetCellLocation(binding, row1));
                     Object cell2 = getCellValue(new ResultSetCellLocation(binding, row2));
-                    result = DBUtils.compareDataValues(cell1, cell2);
+                    if (cell1 instanceof String && cell2 instanceof String) {
```
Comment (serge-rider) [inline plugins/org.jkiss.dbeaver.ui.editors.data/src/org/jkiss/dbeaver/ui/controls/resultset/ResultSetModel.java:1126]: WE can use simple compareToIgnoreCase here.

**Review by serge-rider — CHANGES_REQUESTED** [review]

**Comment by E1izabeth** [comment]
@vijayap20 would you made the requested changes?

**Review by E1izabeth — APPROVED** [review]

**Review by LonwoLonwo — APPROVED** [review]

**Comment by Matvey16** [comment]
Verified

**Review by serge-rider — APPROVED** [review]

**Pull request merged**